api/src/Repository/ActivityResponsibleRepository.php

```diff
@@ -23,9 +26,14 @@ public function __construct(ManagerRegistry $registry) {
     }
 
     public function filterByUser(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, User $user): void {
+        $activityQry = $this->getEntityManager()->createQueryBuilder();
+        $activityQry->select('a');
+        $activityQry->from(Activity::class, 'a');
+        $activityQry->join(UserCamp::class, 'uc', Join::WITH, 'a.camp = uc.camp');
+        $activityQry->where('uc.user = :current_user');
+
         $rootAlias = $queryBuilder->getRootAliases()[0];
-        $queryBuilder->innerJoin("{$rootAlias}.activity", 'activity');
-        $queryBuilder->innerJoin('activity.camp', 'camp');
-        $this->filterByCampCollaboration($queryBuilder, $user);
+        $queryBuilder->andWhere($queryBuilder->expr()->in("{$rootAlias}.activity", $activityQry->getDQL()));
```

Comment:
Did you find out, why the WHERE-IN construct is faster than continuing the join-chain via `filterByCampCollaboration`, now that we have the `view_user_camps`?

Reply:
My intention was to keep the `$queryBuilder` of ApiPlatform as small as possible. 
That is why I try to place the `Where-In` clause at the top level if possible. 
Here this leads to an `ActivityId in (...)` where-clause.

`flterByCampCollaboration` filters directly at the camp level. 
This could therefore only be used indirectly: 
```sql 
select .. 
from .. 
where ar.activity in ( 
   select a.id 
   from activity a 
   where a.campid in ( 
      select .. 
      from .. 
   ) 
) 
```

I didn't want that nesting, so I created the filter directly - but also with the `view_user_camps` view (see `UserCamp::class`)